src/sagemaker/inference_recommender/inference_recommender_mixin.py

```diff
@@ -149,12 +148,26 @@ def right_size(
 
         self._init_sagemaker_session_if_does_not_exist()
 
+        model_name = None
```

Comment:
what if the customer has called `model.create()` or https://github.com/aws/sagemaker-python-sdk/blob/2e44a2cca78fe4f6f69a1bf3b7b2f9fb309e964e/src/sagemaker/model.py#L1607 has been called somewhere? model_name should already be set. We should consider that case too right?

Reply:
also if the model has not been created, is there a way where we can just set the model name and reuse it in `deploy()`?

Reply:
got it makes sense

Reply:
sry mistakenly delete below comment:

I think here the model is created temporarily for invoking IR job, so it should be a temp model created by IR which is independent from customer’s created resources. Also it should be properly cleaned up after IR job done.
Customer experience on model/endpoint creation after right_size should be performed by model.deploy() which is decided by them, so might not to create model(reuse this temp model) for customer in right_size but just making recommendations

Reply:
Agreed, we should allocate and release our own resources to be used for IR job and don't use customer's resources under the hood.